### PR TITLE
Add function to load sharepoint file directly into R 

### PR DESCRIPTION
**Pull request by bvreede**
This PR adds a function to load the holy excel sheet directly (via a temporary download that gets deleted immediately). The function can be re-used for other documents on Sharepoint (it uses `rio::import` which is flexible and can be used for many different data formats).

The function can be used without arguments to download the file that is (currently) the Holy Excel Sheet. When using the function with arguments to download another file problems can arise: it does not contain any checks to verify that paths exist, or that URLs are correct. This is currently mentioned as TODO, but it could be added if the reviewer so desires 🙃

Check the function with `data <- read_from_drive()`

### CODE REVIEW

**Comment by liekelotte** [comment]
Hi Barbara! Thanks a lot for optimizing this part :) I get the following error when trying to run `read_from_drive`:

```
ds_xlsx <- read_from_drive()

Loading Microsoft Graph login for default tenant
Error in login$get_sharepoint_site(site_url, site_id) : 
  object 'ms_site' not found
```

File: R/read_from_drive.R [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#' @return an imported R object
+#' @export
+read_from_drive <- function(path = "General/Digital Skills Workshops 2022.xlsx",
+                            drive = "https://nlesc.sharepoint.com/sites/instructors"){
+
+  #TODO validate path and drive arguments
+  # e.g. path needs to have an extension
+  # drive needs to be sharepoint site and include https://
+
+  # connect to drive
```
Comment (liekelotte) [inline R/read_from_drive.R:26]: Hi Barbara! Thanks a lot for optimizing this part :) I get the following error when trying to run `read_from_drive`:

```
ds_xlsx <- read_from_drive()

Loading Microsoft Graph login for default tenant
Error in login$get_sharepoint_site(site_url, site_id) : 
  object 'ms_site' not found
```

ms_site is a Microsoft365R function, so somewhere this is not imported :)

**Review by liekelotte — CHANGES_REQUESTED** [review]

**Comment by bvreede** [comment]
OK I'm pretty sure I broke all kinds of rules here but.. Can you check if this works now?

```
remotes::install_github(repo = "esciencecenter-digital-skills/training-infrastructure@streamline_microsoft365_use", force=T)
library(traininginfrastructure)
read_from_drive()
```

**Comment by bvreede** [comment]
Can you check what version of `Microsoft365R` you have?
```
packageVersion("Microsoft365R") 
```

**Comment by liekelotte** [comment]
> @liekelotte Can you check what version of `Microsoft365R` you have?
> 
> ```
> packageVersion("Microsoft365R") 
> ```

`2.3.0`!

**Comment by liekelotte** [comment]
> 

this still gives the same error, unfortunately

**Comment by liekelotte** [comment]
> > @liekelotte Can you check what version of `Microsoft365R` you have?
> > ```
> > packageVersion("Microsoft365R") 
> > ```
> 
> `2.3.0`!

Updating worked :D

**Comment by bvreede** [comment]
Niiiiiice!
Please let me know if you're OK with this, and then we can merge!

File: R/read_from_drive.R (hunk 1 above)
Comment (liekelotte) [reply R/read_from_drive.R:23]: this is now fixed

**Review by liekelotte — APPROVED** [review]
Thanks a lot! Really glad we got this fixed :)

**Pull request merged**